### DISCUSSION TITLE
From fastapi.tiangolo.com/async/ (In A Hurry section) Says to only use async if not waiting, (Asynchronous Code section) but async is supposedly for when waiting

### DISCUSSION THREAD

**Discussion by joshb019919** [fastapi/fastapi#6626] [Questions] [answered]
### First check

* [X] I added a very descriptive title to this issue.
* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [X] I already read and followed all the tutorial in the docs and:
&nbsp;&nbsp;&nbsp;&nbsp; ### found something confusing
* [x] I already checked and it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [X] I already checked and it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [X] I already checked and it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
&nbsp;&nbsp;&nbsp; Done

### Confusion

In the tutorial/docs, at the page listed in the title, I find the following:

>If your application (somehow) doesn't have to communicate with anything else and wait for it to respond, use async def.

But further down (plus what I've learned elsewhere about async) it says:

>Asynchronous code just means that the language has a way to tell the computer / program that at some point in the code, it will have to wait for something else to finish somewhere else.

Aren't these contradictions?

In my current dilemma, I'm trying to get my program to connect with IEX APIS to get stock quotes.  I may not absolutely _need_ coroutines for what I'm doing, but I want to demonstrate a more professional undertaking than I have been, of late.

# There was no code to go with this

Idea for using &nbsp to make tabs in markdown by alegscogs and Scott C Wilson:
https://stackoverflow.com/questions/6046263/how-to-indent-a-few-lines-in-markdown-markup

**Comment by joshb019919** [comment, marked as answer]
I believe I understand what went wrong.  Now that I've woken up more and reread the passage, I see that functions only need to be synchronous if connecting to a library which might block the process.  Otherwise, asynchronicity is the way to go.

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue 👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
